SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/TetrahedronFEMForceField.inl

```diff
@@ -1779,7 +1786,8 @@ void TetrahedronFEMForceField<DataTypes>::draw(const core::visual::VisualParams*
     Real minVMN = (Real)1e20, maxVMN = (Real)-1e20;
     helper::ReadAccessor<Data<helper::vector<Real> > > vM =  _vonMisesPerElement;
     helper::ReadAccessor<Data<helper::vector<Real> > > vMN =  _vonMisesPerNode;
-    if (_computeVonMisesStress.getValue() > 0) 
+
+    if ( isVonMisesStressRenderingOptionValid() )
```

Comment:
For me there is an issue here. You have to consider 3 scenario: 

1. user want to visualize vonMisesStress so he set to true ```showVonMisesStressPerNode```  and/or another option that doesn't exist: ```showVonMisesStressPerElement```. If true it should check the value of  computeVonMisesStress (which can be set to 1 or 2). If define by user, use the define value.
2. Same, the user want to show VonMises but compute is not set. Then your check method should turn it to 1 or 2 depending on the numerical method used.
3. The user only want to compute the VonMises (to use it outside, in another component or dump it). Then the compute should be done but the drawing here should not be automatic if "_show.... " is false.